### PR TITLE
TF-2613 Warn for forwards to other domains

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2613 

## Resolved 

https://github.com/linagora/tmail-flutter/assets/80730648/1a6e8c0b-fbe3-4e83-8d61-678148c347d8

### CODE REVIEW

**Comment by hoangdat** [comment]
@chibenwa please share us the message in French also
`you are redirecting emails to another domain. This could be a secuirty threat or considered illegal data extraction. Do you want to proceed?`

**Comment by chibenwa** [comment]
Vous redirigez vos e-mails vers un autre domaine. Cela pourrait constituer une menace pour la sécurité ou être considéré comme une extraction illégale de données. Voulez-vous poursuivre?

File: lib/features/email/presentation/utils/email_utils.dart [hunk 1]
```diff
@@ -49,4 +50,13 @@ class EmailUtils {
       return null;
     }
   }
+
+  static bool isSameDomain({
+    required String emailAddress,
+    required String serverDomain
```
Comment (hoangdat) [inline lib/features/email/presentation/utils/email_utils.dart:56]:
```suggestion
    required String internalDomain
```

File: lib/features/home/domain/extensions/session_extensions.dart [hunk 2]
```diff
@@ -43,4 +45,14 @@ extension SessionExtensions on Session {
       return apiUrl.toString();
     }
   }
+
+  String get serverDomain {
+    if (GetUtils.isEmail(username.value)) {
+      return username.value.split('@').last;
+    } else if (GetUtils.isEmail(personalAccount.name.value)) {
```
Comment (hoangdat) [inline lib/features/home/domain/extensions/session_extensions.dart:52]: `personalAccount` getter maybe throw exception, please handle this case also

File: lib/features/manage_account/presentation/forward/widgets/autocomplete_contact_text_field_with_tags.dart [hunk 3]
```diff
@@ -317,4 +325,73 @@ class _AutocompleteContactTextFieldWithTagsState extends State<AutocompleteConta
...
+      return;
+    }
+
+    final validateSameDomain = EmailUtils.isSameDomain(
+      emailAddress: emailAddress.emailAddress,
+      serverDomain: widget.serverDomain
+    );
+
+    if (!validateSameDomain) {
+      _showWarningDialogWithExternalDomain(
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/forward/widgets/autocomplete_contact_text_field_with_tags.dart:343]: no need await here?

File: lib/features/home/domain/extensions/session_extensions.dart (hunk 2 above)
Comment (hoangdat) [reply lib/features/home/domain/extensions/session_extensions.dart:52]: writing test for this function also

File: lib/features/manage_account/presentation/forward/widgets/autocomplete_contact_text_field_with_tags.dart (hunk 3 above)
Comment (dab246) [reply lib/features/manage_account/presentation/forward/widgets/autocomplete_contact_text_field_with_tags.dart:343]: No, It already has a callback function, we just need to return after calling it to not run the next commands

File: lib/features/email/presentation/utils/email_utils.dart (hunk 1 above)
Comment (dab246) [reply lib/features/email/presentation/utils/email_utils.dart]: done

File: lib/features/home/domain/extensions/session_extensions.dart (hunk 2 above)
Comment (dab246) [reply lib/features/home/domain/extensions/session_extensions.dart]: done

**Review by hoangdat — APPROVED** [review]

**Review by tddang-linagora — APPROVED** [review]

**Pull request merged**